### PR TITLE
COP-3901: Update tests with new email confirmation work-flow

### PR DESCRIPTION
**Pull request by Bothees**
## Description
This PR has changes to update all e2e tests with the new work-flow for registration through email-confirmation.
Added MailSlurp service to create inbox to receive the confirmation email to extract the token to activate the account. 
Updated Login through API call.
Removed tests related to 2FA.

### How to run the tests?
npm run cypress:test:local

## Developer Checklist

\* Required

- [x] Up-to-date with master branch? \*

- [x] Does it have tests?

- [x] Pull request URL linked to JIRA ticket? \*

- [x] All reviewer comments resolved/answered? \*

Ensure you delete the branch once the Pull Request is merged.

### CODE REVIEW

**Review by ajames14 — COMMENTED** [review]
LGTM!

**Review by ajames14 — DISMISSED** [review]

File: cypress/support/commands.js [hunk 1]
```diff
@@ -1,7 +1,11 @@
 const faker = require('faker');
+const { MailSlurp } = require('mailslurp-client');
 
 const { getFutureDate, getPastDate, terminalLog } = require('./utils');
 
+const apiKey = 'e6a3b1f5d1877486bba59baa6800bcccc252997dc7561e56065a0a0d953f75bd';
```
Comment (rafenden) [inline cypress/support/commands.js:6]: This should be an env var.

File: cypress/support/commands.js [hunk 2]
```diff
@@ -83,15 +82,21 @@ Cypress.Commands.add('registerUser', () => {
...
-          `${apiServer}/submit-verification-code`,
-          {
-            email: response.body.email,
-            twoFactorToken: response.body.twoFactorToken,
-          },
-        ).then((response2) => {
-          expect(response2.status).to.eq(200);
+        cy.waitForLatestEmail('cb52ca14-0d64-4821-9970-4e81b47e13c1').then((mail) => {
+          assert.isDefined(mail);
+          token = /token=([A-Za-z0-9-_=]+\.[A-Za-z0-9-_=]+\.?[A-Za-z0-9-_.+/=]*)/.exec(mail.body)[1];
```
Comment (rafenden) [inline cypress/support/commands.js:87]: This var could be initialized in the same line
```suggestion
          const token = /token=([A-Za-z0-9-_=]+\.[A-Za-z0-9-_=]+\.?[A-Za-z0-9-_.+/=]*)/.exec(mail.body)[1];
```

File: cypress/support/commands.js [hunk 3]
```diff
@@ -83,15 +82,21 @@ Cypress.Commands.add('registerUser', () => {
...
+          const email = /email=([A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,})/i.exec(mail.body)[1];
+          cy.log('token:', token, 'email', email);
```
Comment (rafenden) [inline cypress/support/commands.js:89]: No need to log this.

File: cypress/support/commands.js [hunk 4]
```diff
@@ -239,3 +244,11 @@ Cypress.Commands.add('deleteReports', () => {
...
   }
 });
+
+Cypress.Commands.add('createInbox', () => {
+  return mailslurp.createInbox();
+});
+
+Cypress.Commands.add('waitForLatestEmail', (inboxId) => {
+  return mailslurp.waitForLatestEmail(inboxId);
+});
```
Comment (rafenden) [inline cypress/support/commands.js:254]: I don't see the advantage of creating such functions, instead, I would suggest to use mailslurp package directly.

File: package.json [hunk 5]
```diff
@@ -75,9 +75,11 @@
         "file-loader": "^5.1.0",
         "fs-extra": "^9.0.1",
         "jest": "^24.9.0",
+        "mailslurp-client": "^10.0.4",
         "msw": "^0.21.2",
         "node-sass": "^4.14.1",
         "prettier": "^1.19.1",
+        "qs": "^6.9.4",
```
Comment (rafenden) [inline package.json:82]: Please move `qs` package back to the `dependencies` as it's required by the production code.

**Review by rafenden — CHANGES_REQUESTED** [review]

File: cypress/support/commands.js [hunk 6]
```diff
@@ -239,3 +244,11 @@ Cypress.Commands.add('deleteReports', () => {
     cy.exec(query);
   }
 });
+
+Cypress.Commands.add('createInbox', () => {
```
Comment (rafenden) [inline cypress/support/commands.js:248]: This command is never used so please remove it.

**Review by ajames14 — DISMISSED** [review]

File: cypress/support/commands.js (hunk 6 above)
Comment (Bothees) [reply cypress/support/commands.js]: Done.

File: cypress/support/commands.js (hunk 1 above)
Comment (Bothees) [reply cypress/support/commands.js]: Done

File: cypress/support/commands.js (hunk 2 above)
Comment (Bothees) [reply cypress/support/commands.js]: Done.

File: cypress/support/commands.js [hunk 7]
```diff
@@ -3,7 +3,13 @@ const { MailSlurp } = require('mailslurp-client');
...
 const { getFutureDate, getPastDate, terminalLog } = require('./utils');
 
-const apiKey = Cypress.env('mailSlurpApiKey');
+let apiKey;
+
+if (Cypress.env('envname') === 'local') {
+  apiKey = Cypress.env('mailSlurpApiKey');
+} else {
+  apiKey = process.env.MAILSLURP_API_KEY;
+}
```
Comment (rafenden) [inline cypress/support/commands.js:12]: We will fetch it the same way on all environments.
```suggestion
const apiKey = Cypress.env('mailSlurpApiKey');
```

**Review by rafenden — CHANGES_REQUESTED** [review]

**Review by rafenden — APPROVED** [review]

**Pull request merged**